### PR TITLE
[MUGEN][Bugfix] Transformer residual layer

### PR DESCRIPTION
**Pull request by langong347**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #307
* __->__ #305
* #304
* #306
* #303

Summary:
- Fixed a bug that overrode the input to the transformer layer before the latter was added to the attention output
- Updated unit tests

Test Plan:
CI

Differential Revision: [D39420772](https://our.internmc.facebook.com/intern/diff/D39420772)

### CODE REVIEW

**Comment by langong347** [comment]
@langong347 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39420772).

File: torchmultimodal/models/gpt.py [hunk 1]
```diff
@@ -633,16 +633,17 @@ def forward(
...
 
         if use_cache:
             past_key_values = self.attention.cache
 
-        x = x + self.dropout_attn(x)
-        x = self._mlp_block(self.norm_mlp(x))
-        x = x + self.dropout_mlp(x)
+        x = x + self.dropout_attn(hidden)
+
+        hidden = self._mlp_block(self.norm_mlp(x))
```
Comment (ebsmothers) [inline torchmultimodal/models/gpt.py:645]: nit: I think the use of `hidden` here is kinda confusing. Maybe we can more closely follow naming in the [existing transformer layer](https://github.com/facebookresearch/multimodal/blob/main/torchmultimodal/modules/layers/transformer.py#L120-L168)?

**Review by ebsmothers — APPROVED** [review]

File: torchmultimodal/models/gpt.py (hunk 1 above)
Comment (langong347) [reply torchmultimodal/models/gpt.py:645]: Here I am also exposing the option to return the attention probabilities so it's not as straightforward to chain the outputs like in what you quoted (where `return_attn_weights` is always `False`, [ref](https://github.com/facebookresearch/multimodal/blob/main/torchmultimodal/modules/layers/transformer.py#L95)).

I am renaming `hidden` to `attn_hidden_states` and `mlp_hidden_states` to be clear.

**Comment by langong347** [comment]
@langong347 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39420772).

**Pull request closed without merging**